src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(blocks[2][1], gridBagConstraints);
+
+		blocks[2][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(blocks[2][2], gridBagConstraints);
+	}
+
+	public void populatePanel(GameSession gameSession, boolean isRefresh) {
+
+		this.gameSession = gameSession;
+		this.selectedCell = null;
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][1].populate(gameSession.getGameBoard().getBlock(1),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][2].populate(gameSession.getGameBoard().getBlock(2),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[1][0].populate(gameSession.getGameBoard().getBlock(3),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][1].populate(gameSession.getGameBoard().getBlock(4),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][2].populate(gameSession.getGameBoard().getBlock(5),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[2][0].populate(gameSession.getGameBoard().getBlock(6),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][1].populate(gameSession.getGameBoard().getBlock(7),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][2].populate(gameSession.getGameBoard().getBlock(8),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+	}
+	
+	public void populatePencilMark(GameSession gameSession) {
+
+		this.gameSession = gameSession;
+
+		blocks[0][0].populatePencilMark(gameSession.getGameBoard().getBlock(0),
+				gameSession);
+		blocks[0][1].populatePencilMark(gameSession.getGameBoard().getBlock(1),
+				gameSession);
+		blocks[0][2].populatePencilMark(gameSession.getGameBoard().getBlock(2),
+				gameSession);
+
+		blocks[1][0].populatePencilMark(gameSession.getGameBoard().getBlock(3),
+				gameSession);
+		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
+				gameSession);
+		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
+				gameSession);
+
+		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
+				gameSession);
+		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
+				gameSession);
+		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
+				gameSession);
+	}
+
+	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
+		
+		if (null != selectedCell) {

```

Comment:
Again, there are people against and people who likes it. That's why they
even coined the term 'Yoda conditions.'
You don't like it, but I do.

For the argument about 'not English-like' I gave you the Boolean condition
example. Object.equals method itself isn't English-like to begin with. If
Boolean is throwing you off, you can just replace Boolean with String or
any object.

As per the side question regarding Boolean used for conditions, since
Boolean can have three values, there are still valid cases such as following
//=
Boolean a = someService.getSomething();

if (Boolean.TRUE.equals(a)) {
/* true case */
} else if (Boolean.FALSE.equals(a)) {
/* false case */
} else {
/* null case*/
}
//=

while 'if (a) {}' can throw a null pointer exception.

You might run into such checks when you program a web pages in Java with
JSF.









On Mon, Jul 13, 2015 at 3:51 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34503714>:
>
> > +		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
> > +				gameSession);
> > +		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
> > +				gameSession);
> > +
> > +		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
> > +				gameSession);
> > +		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
> > +				gameSession);
> > +		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
> > +				gameSession);
> > +	}
> > +
> > +	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
> > +		
> > +		if (null != selectedCell) {
>
> I'm familiar with the original argument for using it in C/C++, but Java
> and C# do not allow assignment within conditional clauses, so this is
> irrelevant. Even in C/C++, this mistake can be easily prevented with
> const-correctness, testing, code reviews, and just common sense.
>
> As far as readability, this goes against how we think in English, so the
> reader does a double-take on lines like this. Case-in-point: these lines
> immediately stood out to me. Thus I see no advantage to using this style.
>
> Out of curiosity, does Java allow use of the Boolean class in
> conditionals? For example
> Boolean a = Boolean.False; //or just "false" I think will work
> if (a) //dosomething
> ?
>
> If so, I don't see why .equals would ever be necessary.
>
> Here's an article that summarizes my thoughts pretty well from a guy that
> is more familiar with java than I.
>
> http://pushinginertia.com/2011/05/why-yoda-conditions-are-bad-and-usage-of-javas-final-keyword-is-good/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34503714>.
>